### PR TITLE
Fix warnings and deprecations when running unit tests

### PR DESCRIPTION
**Pull request by ausi**

### CODE REVIEW

File: core-bundle/src/Routing/Matcher/LegacyMatcher.php [hunk 1]
```diff
@@ -102,6 +100,8 @@ public function matchRequest(Request $request): array
             $input->setGet('language', $locale);
         }
 
+        @trigger_error('Using the "getPageIdFromUrl" hook has been deprecated and will no longer work in Contao 5.0.', E_USER_DEPRECATED);
+
```
Comment (m-vo) [inline core-bundle/src/Routing/Matcher/LegacyMatcher.php:104]: Why did you move this down?

Comment (m-vo) [reply core-bundle/src/Routing/Matcher/LegacyMatcher.php:104]: Ah, I see `ResourceNotFoundException` could be thrown before.

**Review by m-vo — APPROVED** [review]

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.
